code/script.js

```diff
@@ -271,55 +289,97 @@ const filterCharacters = (keep) => {
+    } 
   } else if (category === 'other') {
+    if (keep) {
+      alert (
+        `Yes, the person has ${value}! Keep all people that has ${value}`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    
     // Similar to the one above
   } else {
+    alert(
+    `No, the person doesn't have ${value}! Remove all people that have ${value}`
+    )
+    charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+  }
+} else if(category === 'hair') {
     if (keep) {
+      alert(
+        `Yes, the person has ${value}! Keep all people that has ${value}`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(
+        `No, the person doesn't have ${value}! Remove all people that have ${value}`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+  
       // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+   } else  if (category === 'eyes') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value}! Keep all people that has ${value}`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
     } else {
+      alert(
+        `No, the person doesn't have ${value}! Remove all people that have ${value}`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
       // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
 
+generateBoard()
+  }
+  
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
+  const playerGuess = confirm(`Do you really think it's ${personToConfirm}?`)
+  if (playerGuess) {
+  checkMyGuess(personToCheck)
   // store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
-}
+}}
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
+  //board.style.display = 'none'
+  //winOrLose.style.display = 'flex'
```

Comment:
These need to be active to hide the board with the characters and for the win or lose-text to show once the guess person is working